odoo/addons/base/tests/test_reports.py

```diff
@@ -39,3 +45,76 @@ def test_reports(self):
                     Report._render_qweb_html(report.id, report_records.ids)
             else:
                 continue
+
+class PDFMixin:
+    def assertPdfEqual(self, test_content, truth_path):
+        with file_open(truth_path, 'rb') as truth_file:
+            truth_content = truth_file.read()
+
+        # neuter dates
+        before, _, after = test_content.partition(b'CreationDate')
+        test_content = before + after.partition(b'\n')[2]
+        before, _, after = truth_content.partition(b'CreationDate')
+        truth_content = before + after.partition(b'\n')[2]
+
+        # assert equal
+        if test_content == truth_content:
+            return
+
+        pdf_dir = os.path.join(config['screenshots'], get_db_name(), 'pdfs')
+        os.makedirs(pdf_dir, exist_ok=True)
```

Comment:
Hello @d-fence @Xavier-Do I would like your input on those two lines

So we can now generate PDFs inside of tests and the way we verify that the generated PDF is good is to verify it is like a pre-generated PDF that we stored on disk (like, a PO-approved PDF).

The thing is... in case the test fail because the rendering is different, I would like to offer the developer a way to download the PDF that has been generated. So he can look it up and see what's wrong. As PDF is a binary format, there is hardly any other alternative...

So for now I'm storing the PDFs inside of this new folder, next to `screenshots` and `screencasts`. What's your opinion about it? Would it be possible to make it downloadable on runbot?